smart_contract/executor/derive/src/token.rs

```diff
@@ -63,5 +63,18 @@ fn impl_try_from_permission_token(ident: &syn2::Ident, generics: &syn2::Generics
                     .map_err(::iroha_executor::permission::PermissionTokenConversionError::Deserialize)
             }
         }
+
+        impl #impl_generics ::core::convert::TryFrom<#ident #ty_generics> for ::iroha_executor::data_model::permission::PermissionToken #where_clause {
+            type Error = ::iroha_executor::permission::PermissionTokenConversionError;
+
+            fn try_from(token: #ident #ty_generics) -> ::core::result::Result<Self, Self::Error> {
+                let definition_id = #token_id;
+
+                let payload = ::serde_json::to_value::<#ident #ty_generics>(token)
+                    .map_err(::iroha_executor::permission::PermissionTokenConversionError::Serialize)?;
```

Comment:
I think it's ok to panic here. It's assumed that executor is valid in the sense that permission tokens are valid. An error here means that executor is faulty, and that is not a recoverable error